### PR TITLE
[TT-4571] Fix some edge cases of OAS auth source conversions

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR fixes some edge cases of auth source OAS conversions

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3912/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1917049233)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3912/merge
Commit: 22bd364314b6b3df7053cc15564c207b5d149179 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1917149742)

File: apidef/oas/security.go [hunk 1]
```diff
@@ -122,15 +122,15 @@ func (s *OAS) fillSecurityScheme(ac *apidef.AuthConfig) {
 	var loc, name string
 
 	switch {
-	case ref.Value.In == header || (ref.Value.In == "" && !ac.DisableHeader):
+	case ref.Value.In == header || (ref.Value.In == "" && ac.AuthHeaderName != ""):
 		loc = header
 		name = ac.AuthHeaderName
 		ac.AuthHeaderName = ""
-	case ref.Value.In == query || (ref.Value.In == "" && ac.UseParam):
+	case ref.Value.In == query || (ref.Value.In == "" && ac.ParamName != ""):
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply apidef/oas/security.go:129]: Yes because user may disable it but there can be a name value set in the scheme.

**Review by urbanishimwe — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: c09874608ffc3b4cf48b82173725099044ba63e2 Fix some edge cases of OAS auth source conversions (#3912)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1924672249)
